src/System.Runtime.Caching/src/System/Runtime/Caching/Dbg.cs

```diff
@@ -826,13 +832,23 @@ internal static bool IsTagPresent(string tagName)
         internal static void Break()
         {
 #if DEBUG
-            if (NativeMethods.IsDebuggerPresent()) {
-                NativeMethods.DebugBreak();
-            }
-            else if (!Debugger.IsAttached) {
-                Debugger.Launch();
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                if (NativeMethods.IsDebuggerPresent()) 
+                {
+                    NativeMethods.DebugBreak();
```

Comment:
What's the purpose of these NativeMethods?  Why not just remove them?  Or if for some reason we need it, why not just add the IsOSPlatform &&'d into this condition and avoid having the duplicated Debugger.Break calls?

Reply:
Thanks @stephentoub for the comment. I will change it with 'IsOSPlatform &&' conditions as I also struggled a bit with creating the code duplication :)

This whole DBG class came from some old ages when the .NET framework didn't have enough managed equivalents of win32 APIs and the cache code was raw and required tracing and debugging. The cache code has matured since then and I don't remember any instance of when any of these checks were used last time. So to be honest it might be beneficial to just get rid of it all (or at least get rid of NativeMethods). But then we'd lose some of the checks (and some of pretty much never used functionality like monitoring registry for changes, etc)... Given that this was debug code only and there might be a slight possibility that someone might need to use it in the future, I decided to keep the DBG class as is.

Does merely referencing Windows dlls in NativeMethods cause the dlls to get mapped to the process memory (I'd assume most of them are already mapped most of the time)?

Reply:
Ok, thanks for the explanation.  At some point I suggest we remove the arguably dead code rather than continuing to maintain it.

> Does merely referencing Windows dlls in NativeMethods cause the dlls to get mapped to the process memory

With JIT'd code, generally no.